plugins/infinite-sky.cpp

```diff
@@ -0,0 +1,270 @@
+                // fine
+                df::map_block_column *column = world->map.column_index[a][b];
+                if (!column) {
+                    DEBUG(cycle, out)
+                        .print("%s, line %d: column is null (%d, %d).\n",
+                               __FILE__, __LINE__, a, b);
+                    continue;
+                }
+                df::map_block_column::T_unmined_glyphs *glyphs =
+                    new df::map_block_column::T_unmined_glyphs;
+                glyphs->x[0] = 0;
+                glyphs->x[1] = 1;
+                glyphs->x[2] = 2;
+                glyphs->x[3] = 3;
+                glyphs->y[0] = 0;
+                glyphs->y[1] = 0;
+                glyphs->y[2] = 0;
+                glyphs->y[3] = 0;
+                glyphs->tile[0] = 'e';
+                glyphs->tile[1] = 'x';
+                glyphs->tile[2] = 'p';
+                glyphs->tile[3] = '^';
+                column->unmined_glyphs.push_back(glyphs);
+            }
+        }
+    }
+
+    // Update global z level flags
+    df::z_level_flags *flags = new df::z_level_flags[z_count_block + howMany];
+    memcpy(flags, world->map_extras.z_level_flags,
+           z_count_block * sizeof(df::z_level_flags));
+    for (int32_t count = 0; count < howMany; count++) {
+        flags[z_count_block + count].whole = 0;
+        flags[z_count_block + count].bits.update = 1;
+    }
+    world->map.z_count_block += howMany;
+    world->map.z_count += howMany;
+    delete[] world->map_extras.z_level_flags;
+    world->map_extras.z_level_flags = flags;
+}
+
+struct infinitesky_options {
+    // whether to display help
+    bool help = false;
+
+    // how many z levels to generate immediately (0 for none)
+    int32_t n = 0;
+
+    static struct_identity _identity;
+};
+static const struct_field_info infinitesky_options_fields[] = {
+    {struct_field_info::PRIMITIVE, "help", offsetof(infinitesky_options, help), &df::identity_traits<bool>::identity, 0, 0},
+    {struct_field_info::PRIMITIVE, "n", offsetof(infinitesky_options, n), &df::identity_traits<int32_t>::identity, 0, 0}
+};
+struct_identity infinitesky_options::_identity{sizeof(infinitesky_options), &df::allocator_fn<infinitesky_options>, NULL, "infinitesky_options", NULL, infinitesky_options_fields};
+
+command_result infiniteSky(color_ostream &out,
+                           std::vector<std::string> &parameters) {
+    if (!Core::getInstance().isMapLoaded() || !World::isFortressMode()) {
+        out.printerr("Cannot run %s without a loaded fort.\n", plugin_name);
+        return CR_FAILURE;
+    }
+
+    infinitesky_options opts;
+    if (!Lua::CallLuaModuleFunction(out, "plugins.infinite-sky",
+                                    "parse_commandline",
+                                    std::make_tuple(&opts, parameters)) ||
+        opts.help)
+        return CR_WRONG_USAGE;
+
+    if (opts.n != 0) {
```

Comment:
```suggestion
    if (opts.n > 0) {
```

Reply:
makes the semantics a bit clearer (even though negative numbers can't happen with current code)